src/cmd/singularity/cli/actions.go

```diff
@@ -17,51 +17,131 @@ import (
-      $ singularity shell /tmp/Debian.img
-      Singularity/Debian.img> pwd
-      /home/gmk/test
-      Singularity/Debian.img> exit
-      
-      $ singularity shell -C /tmp/Debian.img
-      Singularity/Debian.img> pwd
-      /home/gmk
-      Singularity/Debian.img> ls -l
-      total 0
-      Singularity/Debian.img> exit
-      
-      $ sudo singularity shell -w /tmp/Debian.img
-      $ sudo singularity shell --writable /tmp/Debian.img
-      
-      $ singularity shell instance://my_instance 
-      
-      $ singularity shell instance://my_instance
-      Singularity: Invoking an interactive shell within container...
-      Singularity container:~> ps -ef
-      UID        PID  PPID  C STIME TTY          TIME CMD
-      ubuntu       1     0  0 20:00 ?        00:00:00 /usr/local/bin/singularity/bin/sinit
-      ubuntu       2     0  0 20:01 pts/8    00:00:00 /bin/bash --norc
-      ubuntu       3     2  0 20:02 pts/8    00:00:00 ps -ef`
+$ singularity shell /tmp/Debian.img
+Singularity/Debian.img> pwd
+/home/gmk/test
+Singularity/Debian.img> exit
+
+$ singularity shell -C /tmp/Debian.img
+Singularity/Debian.img> pwd
+/home/gmk
+Singularity/Debian.img> ls -l
+total 0
+Singularity/Debian.img> exit
+
+$ sudo singularity shell -w /tmp/Debian.img
+$ sudo singularity shell --writable /tmp/Debian.img
+
+$ singularity shell instance://my_instance 
+
+$ singularity shell instance://my_instance
+Singularity: Invoking an interactive shell within container...
+Singularity container:~> ps -ef
+UID        PID  PPID  C STIME TTY          TIME CMD
+ubuntu       1     0  0 20:00 ?        00:00:00 /usr/local/bin/singularity/bin/sinit
+ubuntu       2     0  0 20:01 pts/8    00:00:00 /bin/bash --norc
+ubuntu       3     2  0 20:02 pts/8    00:00:00 ps -ef`
+
+var runUse string = `run [run options...] <container>`
+
+var runShort string = `execute the commands in a container runscript`
+
+var runLong string = `
+This command will launch a Singularity container and execute a runscript
+if one is defined for that container. The runscript is a metadata file within
+the container that containes shell commands. If the file is present (and 
+executable) then this command will execute that file within the container 
+automatically. All arguments following the container name will be passed 
+directly to the runscript.
+
+singularity run accepts the following container formats:` + formats
 
 var runExamples string = `
+# Here we see that the runscript prints "Hello world: "
+$ singularity exec /tmp/Debian.img cat /singularity
+#!/bin/sh
+echo "Hello world: "
+
+# It runs with our inputs when we run the image
+$ singularity run /tmp/Debian.img one two three
+Hello world: one two three
+
+# Note that this does the same thing
+$ ./tmp/Debian.img one two three
 `
 
 func init() {
 	actionCmds := []*cobra.Command{
-		execCmd,
-		shellCmd,
-		runCmd,
+		ExecCmd,
+		ShellCmd,
+		RunCmd,
+	}
+
+	// TODO : the next n lines of code are repeating too much but I don't
```

Comment:
So *if* we decide to stick with this approach of running the man pages when a user calls help, this is not a great way to define these help functions. Within the [cobra.Command](https://godoc.org/github.com/spf13/cobra#Command) there is a field `Command.Annotations map[string]string`. This field allows (essentially) arbitrary meta data to be embedded within a `cobra.Command` type. So if (again, big if) we decide to stick with this approach, we can do something like this within the initialization of the `.Command` structures. 

```
var ExecCmd = &cobra.Command{
	DisableFlagsInUseLine: true,
 	Args: cobra.MinimumNArgs(2),
 	Run: func(cmd *cobra.Command, args []string) {
 		a := append([]string{"/.singularity.d/actions/exec"}, args[1:]...)
 		execWrapper(cmd, args[0], a)
 	},

	Use:     execUse,
	Short:   execShort,
	Long:    execLong,
 	Example: execExamples,
        Annotations: map[string]string{
                "manstring": "singularity-exec",
        },
 }
```

Then a general help function can access `c.Annotations["manstring"]` to get the man string for that specific command.